src/Core/src/Layouts/LayoutExtensions.cs

```diff
@@ -45,6 +45,11 @@ public static Rect ComputeFrame(this IView view, Rect bounds)
 			{
 				consumedWidth = Math.Min(bounds.Width, view.MaximumWidth);
 			}
+			else if (!IsExplicitSet(view.Width))
```

Comment:
@kubaflo thanks for fixing this, I guess code owners are going to ask you to write some unit tests on this.
Assuming the code you proposed does not create regressions in other situations, I think we can avoid computing `IsExplicitSet` twice.

```cs
if (!IsExplicitSet(view.Width))
{
    var desiredWidth = view.HorizontalLayoutAlignment == LayoutAlignment.Fill ? view.MaximumWidth : consumedWidth;
    consumedWidth = Math.Min(bounds.Width, desiredWidth);
}
```